src/integrationTest/java/org/opensearch/security/http/AuthInfo.java

```diff
@@ -18,13 +18,25 @@
 class AuthInfo {
 
 	private final List<String> customAttributeNames;
+	private final boolean isInternal;
+	private final String authDomain;
 
-	@ConstructorProperties("custom_attribute_names")
-	public AuthInfo(List<String> customAttributeNames) {
+	@ConstructorProperties({"custom_attribute_names", "user_is_internal", "user_auth_domain"})
+	public AuthInfo(List<String> customAttributeNames, boolean isInternal, String authDomain) {
 		this.customAttributeNames = customAttributeNames;
+		this.isInternal = isInternal;
+		this.authDomain = authDomain;
 	}
 
 	public List<String> getCustomAttributeNames() {
 		return customAttributeNames;
 	}
+
+	public boolean isInternal() {
+		return isInternal;
+	}
```

Comment:
This seems redundant with the auth domain, lets remove it, what do you think?

Reply:
This class deserializes the response from `/authinfo` for testing, its driven by the annotation above the constructor. This is needed to get the value from the response.

Reply:
What does `isInternal` mean when it is true vs false?  Seems like it indicates that the user is associated with the internal auth domain, but it isn't used to change a call pattern elsewhere in the security plugin.

Reply:
That's exactly what it means. In order to accomplish the security model for scheduled jobs where we can lookup authz from user principal alone, then this information would need to be available when persisting the identity associated with a scheduled job. For extensions, I was thinking that this information could be embedded in the obo token that would be presented by an extension when registering a scheduled job.

Reply:
> for testing

If it is just for testing can't we check for the internal auth provider type?  Its an extra object to serialize and there is a tax on performance for each field.

Reply:
This whole class is for testing, there is no performance impact

Reply:
I commented on the wrong class - you are right this class is fine, but I'm going to move my comment to the correct file